SMLHelper/Options/Attributes/ButtonAttribute.cs

```diff
@@ -0,0 +1,49 @@
+﻿namespace SMLHelper.V2.Options.Attributes
+{
+    using Json;
+    using System;
+    using QModManager.Utility;
+
+    /// <summary>
+    /// Attribute used to signify the attrubuted method should be represented in the mod's options menu
+    /// as a <see cref="ModButtonOption"/>.
+    /// When the button is clicked, the given method will run.
+    /// </summary>
+    /// <example>
+    /// <code>
+    /// using SMLHelper.V2.Json;
+    /// using SMLHelper.V2.Options;
+    /// using QModManager.Utility;
+    /// 
+    /// [Menu("My Options Menu")]
+    /// public class Config : ConfigFile
+    /// {
+    ///     [Button("My Cool Button")]
+    ///     public static void MyCoolButton(object sender, ButtonClickedEventArgs e)
+    ///     {
+    ///         Logger.Log(Logger.Level.Info, "Button was clicked!");
+    ///         Logger.Log(Logger.Level.Info, e.Id.ToString());
+    ///     }
+    /// }
+    /// </code>
+    /// </example>
+    /// <seealso cref="MenuAttribute"/>
+    /// <seealso cref="ButtonClickedEventArgs"/>
+    /// <seealso cref="Logger"/>
+    /// <seealso cref="ConfigFile"/>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
+    public sealed class ButtonAttribute : ModOptionAttribute
+    {
+        /// <summary>
+        /// Signifies the attributed method should be represented in the mod's options menu as a <see cref="ModButtonOption"/>
+        /// and sets its label.
+        /// </summary>
+        /// <param name="label">The label for the button. If none is set, the name of the method will be used.</param>
+        public ButtonAttribute(string label = null) : base(label) { }
+
+        /// <summary>
+        /// Signifies the attributed method should be represented in the mod's options menu as a <see cref="ModButtonOption"/>.
+        /// </summary>
+        public ButtonAttribute() { }
```

Comment:
There appears to be some ambiguity between these constructors.  
One is _parameterless_, the other has a single, _optional_ parameter.

_The same happens across many other attribute classes here_

Since the base class `ModOptionAttribute` only has a single constructor with an optional parameter, do these child classes need to have a default constructor?

Reply:
Found `new TAttribute()`

Reply:
Yup, was about to say. It's currently solely because of this constraint, because we need/want to be able to create a default instance in some cases. I wish that `new()` took optional parameters into consideration.